lib/ansible/module_utils/facts/hardware/__init__.py

```diff
@@ -0,0 +1,32 @@
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+from ansible.module_utils.facts.hardware import aix
+from ansible.module_utils.facts.hardware import darwin
+from ansible.module_utils.facts.hardware import dragonfly
+from ansible.module_utils.facts.hardware import freebsd
+from ansible.module_utils.facts.hardware import hpux
+from ansible.module_utils.facts.hardware import hurd
+from ansible.module_utils.facts.hardware import linux
+from ansible.module_utils.facts.hardware import netbsd
+from ansible.module_utils.facts.hardware import openbsd
+from ansible.module_utils.facts.hardware import sunos
```

Comment:
That may be a little trickier. Currently the __init__.py module is used to import all of the hardware.Hardware() based impls (the metaclass based ones from current facts). If the modules aren't imported, the __new__() doesnt find the subclasses via module_utils.basic.get_all_subclasses() so only the default Hardware() is found.

I have some ideas on how to eliminate the metaclass use, but not sure its worthwhile yet.

The main downside to the explicit imports is it means all of the modules are loading when the facts package is imported, which makes ansiballz include all of the modules in its import list.

I have some ideas on how to do that an ansiballz construction time instead of (or in addition to) module runtime. But that is also a good chunk of 'facts plugins' which isnt on the 2.4 roadmap